src/NSwag.AspNetCore.Launcher/Program.cs

```diff
@@ -0,0 +1,178 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Reflection;
+
+namespace NSwag.AspNetCore.Launcher
+{
+    public class Program
+    {
+        private const string EntryPointType = "NSwag.SwaggerGeneration.AspNetCore.AspNetCoreToSwaggerGeneratorCommandEntryPoint";
+        private static readonly AssemblyName AspNetCoreSwaggerGenerationAssembly = new AssemblyName("NSwag.SwaggerGeneration.AspNetCore");
+        private static readonly Version NSwagVersion = typeof(Program).GetTypeInfo().Assembly.GetName().Version;
+        // List of assemblies and versions referenced by NSwag.SwaggerGeneration.AspNetCore. This represents the minimum versions
+        // required to successfully run the tool.
+        private static readonly Dictionary<string, AssemblyLoadInfo> NSwagReferencedAssemblies = new Dictionary<string, AssemblyLoadInfo>(StringComparer.OrdinalIgnoreCase)
```

Comment:
cc @rynowak This is the TPA list thing. The tool attempts to load versions referenced in the application first and throws if the application uses a version lower than the minimum requirement.